File: src/yandex/coderun/Problem280.java
package yandex.coderun;

import java.io.*;
import java.util.Arrays;
import java.util.Comparator;

public class Problem280 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(reader.readLine());
        int[][] arr = new int[n][2];
        for (int i = 0; i < n; i++) {
            String s = reader.readLine();
            int cPos = s.indexOf(' ');
            int a = Integer.parseInt(s, 0, cPos, 10);
            int b = Integer.parseInt(s, cPos + 1, s.length(), 10);
            arr[i][0] = Math.max(a, b);
            arr[i][1] = Math.min(a, b);
        }
        Arrays.sort(arr, Comparator.comparingInt(a -> a[0]));

        int ans = 0, maxHeight = 0;
        for (int i = n - 1, j = n - 1; i >= 0; i--) {
            if (arr[i][1] < maxHeight) continue;
            while (arr[i][0] < arr[j][0]) maxHeight = Math.max(maxHeight, arr[j--][1]);
            if (arr[i][1] >= maxHeight) ans++;
        }
        writer.write(Integer.toString(ans));
        reader.close();
        writer.close();
    }
}

File: src/yandex/coderun/Problem280.kt
package yandex.coderun

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.lang.Integer.parseUnsignedInt
import java.util.*
import kotlin.math.max
import kotlin.math.min

fun main() {
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val writer = BufferedWriter(OutputStreamWriter(System.out))
    val n = reader.readLine().toInt()
    val arr = Array(n) { IntArray(2) }
    for (i in 0 until n) {
        val s = reader.readLine()
        val cPos = s.indexOf(' ')
        val a = parseUnsignedInt(s, 0, cPos, 10)
        val b = parseUnsignedInt(s, cPos + 1, s.length, 10)
        arr[i][0] = max(a, b)
        arr[i][1] = min(a, b)
    }
    writer.write(solve(arr).toString())
    reader.close()
    writer.close()
}

fun solve(arr: Array<IntArray>): Int {
    val n = arr.size
    Arrays.sort(arr) { a, b -> a[0] - b[0] }
    var ans = 0
    var maxHeight = 0
    var j = n - 1
    for (i in n - 1 downTo 0) {
        if (arr[i][1] < maxHeight) continue
        while (arr[i][0] < arr[j][0]) maxHeight = max(maxHeight, arr[j--][1])
        if (arr[i][1] >= maxHeight) ans++
    }
    return ans
}